pkg/cmd/dcos_cluster_remove.go

```diff
@@ -9,16 +10,38 @@ import (
 
 // newCmdClusterRemove removes a cluster.
 func newCmdClusterRemove(ctx *cli.Context) *cobra.Command {
+	var removeAll bool
 	cmd := &cobra.Command{
 		Use:  "remove",
 		Args: cobra.MaximumNArgs(1),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if removeAll && len(args) == 1 {
+				return errors.New("cannot accept both a cluster name and the --all option")
+			}
+			if !removeAll && len(args) == 0 {
+				return errors.New("either a cluster name or the --all option must be passed")
+			}
+			return nil
+		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			conf, err := ctx.ConfigManager().Find(args[0], false)
-			if err != nil {
-				return err
+			// Remove a single cluster.
+			if len(args) == 1 {
```

Comment:
Having an `if removeAll {` to handle the two cases would be more straightforward here.